plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/embeddings/MethodSignatureEmbedding.kt

```diff
@@ -15,6 +18,7 @@ import org.jetbrains.kotlin.formver.viper.ast.*
  * Example: Foo.bar(x: Int) --> Foo$bar(this: Foo, x: Int)
  */
 class MethodSignatureEmbedding(
+    symbol: FirFunctionSymbol<*>,
```

Comment:
By "computing all this stuff in `ProgramConverter`" you mean not passing the symbol to the contructor and computing the pre and postconditions in `ProgramConverter`? The problem with that is that in order to computer the postconditions from the contract you already need the `MethodSignatureEmbedding`.
So you either have to do it as it was done before by passing the condition later into the `toMethod` function, but then you again can't access the postconditions from the `MethodSignatureEmbedding` which was the whole point. Or you have some sort of setter which later sets the postconditions which I find to be less elegant than just passing the symbol to the constructor.
One thing we could do though is not pass a name as it can be computed by `MethodSignatureEmbedding` with the symbol.

Reply:
Ah, I see the chicken and egg problem here, yes, sorry.
I'm not sure what's best... I agree that having *all* the preconditions and postconditions available here is nice, that probably trumps other considerations.
LGTM!